resolvers.py

```diff
@@ -19,11 +21,62 @@ def get_dates_and_price_check(
             **search_criteria
     ) -> List[DatesCheck]:
         validated_search_criteria = DatesAndPriceCheckInputValidator(**search_criteria)
-        results_list = []
+        houses_list = {}
         for listing_id in validated_search_criteria.listing_ids:
-            listing = data[listing_id]
-            dates_check = DatesCheck(
-                listing_id=listing.listing_id
-            )
-            results_list.append(dates_check)
+            houses_list[listing_id] = data[listing_id]
+        return QueryResolvers.getResultsList(validated_search_criteria, houses_list)
+
+    @staticmethod
+    @query.field("ranked_search_results")
+    def get_ranked_search_results(
+            _,
+            info,
+            **search_criteria
+    ) -> List[DatesCheck]:
+        validated_search_criteria = RankedSearchResultInputValidator(**search_criteria)
+        return QueryResolvers.getResultsList(validated_search_criteria, data)
```

Comment:
The entire `getResultsList` function violates OCP. Ask @ChenYuan-Lee for more details. You can afford to move some of the logic in the fn into the `DatesCheck` class.